packages/data/lib/repository/network_repository.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:data/service/api_service.dart';
+import 'package:dio/dio.dart';
+import 'package:domain/use_cases/model/palindrome_response.dart';
+import 'package:domain/use_cases/repository/network_repository.dart';
+
+import 'api_base_repository.dart';
+
+class NetworkRepository extends ApiBaseRepositoryImpl
+    implements INetworkRepository {
+  final ApiService _service;
+  final CancelToken _cancelToken;
+
+  NetworkRepository(
+    this._service,
+    this._cancelToken,
+  ) : super(cancelToken: _cancelToken);
+
+  @override
+  Future<PalindromeResponse> checkPalindrome(String palindrome) {
+    return _service
+        .get(
+          path: '/ispalindrome/$palindrome',
```

Comment:
FYI: it seems better if u'll create a class ApiPath with static method getIsPalindromeRoute. this method will receive a string with palindrome value as a parameter and return concatenated string. just delegating forming path responsibility to the separate class

Reply:
fixed